src/tools/utils.js

```diff
@@ -71,11 +71,20 @@ export function dumpJSON(obj, spacing = 0) {
  * @param {boolean} [allowDelete=false]
  * @returns T
  */
-function processChangedObjectFields(handler, desiredAssetState, currentAssetState, objectFields = [], allowDelete = false) {
+export function processChangedObjectFields({
+  handler, desiredAssetState, currentAssetState, allowDelete = false
+}) {
```

Comment:
Changing the function to accept an object of parameters instead of several arguments. Also, I removed the `objectFields` argument because it can be derived from the handler directly.